sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisErrorSuite.scala

```diff
@@ -787,8 +787,70 @@ class AnalysisErrorSuite extends AnalysisTest {
     val error = intercept[AnalysisException] {
       SimpleAnalyzer.checkAnalysis(join)
     }
-    assert(error.message.contains("Failure when resolving conflicting references in Join"))
-    assert(error.message.contains("Conflicting attributes"))
+    checkError(
+      exception = error,
+      errorClass = "RESOLVED_PLAN_HAVE_CONFLICTING_ATTRS",
+      parameters = Map(
+        "nodeName" -> "Join",
+        "plan" -> "'Join Cross\n:- LocalRelation <empty>, [a#0]\n+- LocalRelation <empty>, [a#0]\n",
+        "conflictingAttributes" -> "\"a\""
+      )
+    )
+  }
+
+  test("error test for self-intersect") {
+    val intersect = Intersect(testRelation, testRelation, true)
+    val error = intercept[AnalysisException] {
+      SimpleAnalyzer.checkAnalysis(intersect)
```

Comment:
@MaxGekk I created test cases use `SimpleAnalyzer.checkAnalysis` directly.

Reply:
@MaxGekk Do you have any suggestion ?

Reply:
I see but if you cannot trigger the error from user space, we should convert the error to an internal one.

Reply:
@beliefer Could you try to trigger it by from SQL.

Reply:
> @beliefer Could you try to trigger it by from SQL.

We can't trigger it by SQL. Let me convert it to internal error.

Reply:
cc @cloud-fan Just to-double check that errors are internal.